### DISCUSSION TITLE
connecting between targets aws_qs and rsconnect

### DISCUSSION THREAD

**Discussion by yonicd** [ropensci/targets#630] [Help]
I am trying to figure out what is the best way to deploy to rsconnect while leveraging the format aws_qs.

I am able to write to my bucket and run my shiny app locally. 

I do this using tar_config because the shiny app isn't in the root dir and running using the RS IDE button sets the `pwd` to where the file active location.

```r
tar_config_set(
    store = file.path(rprojroot::find_root('_targets.R'),'_targets')
)  
```

Obviously, I can't use this same setup for RSC because there is no `_targets.R` to work off of. Which leads me to another problem that there is no  `_targets/meta/meta` in RSC

for now I have an if statement set up that uses `aws.s3` on RSC

```r
get_rsc_aws <- function(object, object_path, bucket_name){
  aws.s3::s3read_using(
    object = file.path(object_path, object), 
    FUN = qs::qread, 
    bucket = bucket_name)  
}
```

What would be the correct way to set this up that I can have the same `tar_load()` call for both environments? would copying to the aws bucket `_targets/meta` solve this?

Thanks

**Comment by wlandau** [comment]
So if I understand correctly, you have a `targets` pipeline that has an `aws_qs` target and deployed a Shiny app with another target? (Or a Shiny app on RStudio Connect that requires an `aws_qs` target created from an otherwise totally independent `targets` pipeline?). `_targets/meta/meta` needs to exist locally in the app in order for `tar_load()` and `tar_read()` to work, so that would either need to be deployed along with the app, or the app would need to pull it. But I like your `aws.s3::s3read_using()` solution. That should definitely work.

**Reply by yonicd** [reply]
I have the latter setup. 

I’m guessing copying the meta subdir to rsc is a headache. Right now there isn’t a resource option for rsconnect in tar_target().
